chichiwang/exercise-js
# Build Animated Hero Banner

## Description
Linked to Issue: #81
Create an animated Hero Banner component to be displayed on the home page. It is designed to be contained within the max width.

This Hero Banner is not designed to be responsive, it is fixed in size, but is designed to be centered within whatever width screen exists. At smaller resolutions, the screen will crop into the center of the banner. At larger resolutions, the banner is maximum constrained to 1920px (the content width constraint for the site).

Banner animations are disabled for print media.

## Changes
* A component `pages/components/HeroBanner/index.tsx` is created
  * Contains sub-components:
    * `pages/components/HeroBanner/SceneBlue/index.tsx`
    * `pages/components/HeroBanner/SceneMaroon/index.tsx`
      * `pages/components/HeroBanner/SceneMaroon/CloudscapeBackground.tsx`
    * `pages/components/HeroBanner/SceneOrange/index.tsx`
      * `pages/components/HeroBanner/SceneOrange/CloudscapeForeground.tsx`
    * `pages/components/HeroBanner/SceneRed/index.tsx`
      * `pages/components/HeroBanner/SceneRed/Cityscape.tsx`
      * `pages/components/HeroBanner/SceneRed/MountainRange.tsx`
      * `pages/components/HeroBanner/SceneRed/TaipeiBuilding.tsx`
* `HeroBanner` component rendered in `pages/Home/index.tsx`

## Steps to QA
* Pull down and switch to this branch
* Run the application and verify the home route `/` in the browser contains the animated banner

![image](https://github.com/chichiwang/tamsui/assets/2304118/7dbcca7b-c7ee-4e48-9ed1-a9cc2ea65871)

* Verify the animation does not run for print media, either by developer tools or print preview

![image](https://github.com/chichiwang/tamsui/assets/2304118/052f9dde-6843-40d3-a682-9a2a23ed30af)
